### PR TITLE
Create Distance in KM Comparison library function

### PR DESCRIPTION
**Pull request by RossKen**
Based off #1038 and #1039 plus a couple of logging and docs edits

### CODE REVIEW

File: tests/test_km_distance_level.py
Comment (ADBond) [inline tests/test_km_distance_level.py:26]:
```suggestion
    print(
        cl.distance_in_km_at_thresholds(
            lat_col="latlong[0]", long_col="latlong[1]", km_thresholds=[1, 5, 10]
        ).as_dict()
    )
    print(
        cl.distance_in_km_at_thresholds(
            lat_col="ll['lat']", long_col="ll['long']", km_thresholds=[1, 5, 10]
        ).as_dict()
    )

@pytest.mark.parametrize(
    ("cll"),
    [
        pytest.param(clld, id="DuckDB Distance in KM cll simple run tests"),
        pytest.param(clls, id="Spark Distance in KM cll simple run tests"),
    ],
)
def test_simple_run_cll(cll):
    print(
        cll.distance_in_km_level(
            lat_col="lat", long_col="long", km_threshold=1
        ).as_dict()
    )
    print(
        cll.distance_in_km_level(
            lat_col="latlong[0]", long_col="latlong[1]", km_threshold=0.1
        ).as_dict()
    )
    print(
        cll.distance_in_km_level(
            lat_col="ll['lat']", long_col="ll['long']", km_threshold=10
        ).as_dict()
    )
```
Checks for using array/object columns - as things stand the `cll` version passes, but the new comparison code does not

**Review by ADBond — CHANGES_REQUESTED** [review]
Looks good - and extended logging functionality useful. Just one point re: object/array columns, and a query about default values

File: splink/comparison_library.py
Comment (RossKen) [reply splink/comparison_library.py]: Pushed

File: splink/comparison_library.py
Comment (RossKen) [reply splink/comparison_library.py:594]: As discussed separately - using `InputColumn` on null level reveals an issue with `_sql_dialect` so we have agreed to leave this PR as is and deal with it separately.

Comment (ADBond) [reply splink/comparison_library.py:594]: Yep sounds good - might be worth commenting out the failing tests for the time being so as not to confuse matters if anyone else is branching of `master`, but I'll leave that up to you

**Review by ADBond — APPROVED** [review]

File: splink/comparison_library.py
Comment (RossKen) [reply splink/comparison_library.py:594]: Yea, that's my bad. Commented out and pushed. Will merge when tests pass.

Comment (RossKen) [reply splink/comparison_library.py:594]: Thanks for all your help!

**Pull request merged**
